pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -804,6 +805,42 @@ func (nc *Controller) monitorNodeHealth() error {
 	return nil
 }
 
+// labelNodeDisruptionExclusion is a label on nodes that controls whether they are
+// excluded from being considered for disruption checks by the node controller.
+const labelNodeDisruptionExclusion = "node-controller.kubernetes.io/exclude-disruption"
+
+func isNodeExcluded(node *v1.Node) bool {
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyNodeRoleBehavior) {
+		if legacyIsMasterNode(node.Name) {
+			return true
+		}
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeDisruptionExclusion) {
+		if _, ok := node.Labels[labelNodeDisruptionExclusion]; ok {
+			return true
+		}
+	}
+	return false
+}
+
+// legacyIsMasterNode returns true if given node is a registered master according
+// to the logic historically used for this function. This code path is deprecated
+// and the node disruption exclusion label should be used in the future.
+// DEPRECATED: Will be removed in 1.19
```

Comment:
add a comment that this should explicitly not be updated to check the node role label

Reply:
Do we have a central doc or exported library on reserved labels?